src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -271,6 +296,10 @@ async def test_TC_MCORE_FS_1_4(self):
         self.step(1)
 
         th_server_th_node_id = 1
+
+        # Wait for TH_SERVER_NO_UID get initialized.
+        await self.wait_for_server_initialization(self.th_server_port)
```

Comment:
Would the end of `setup_class` be better after we initialized the application

Reply:
Wherever this is called, it might also be good to have a try block where the exception for TimeoutError calls `asserts.fail(f"TH_SERVER_NO_UID server failed to open port {self.th_server_port}")`

Reply:
But setup_class can not run await, we need to use asyncio to run the asynchronous server initialization, something like:
    
    asyncio.run(self.wait_for_server_initialization(self.th_server_port))